app/helpers/activity_helper.rb

```diff
@@ -20,4 +20,9 @@ def user_can_edit?(user, activity)
     return true if user.administrator? || user.fund_manager?
     user.delivery_partner? && activity.is_project_level?
   end
+
+  def user_can_add_budget?(user, activity)
```

Comment:
Could we merge this logic into the budget policies and ask the Budget policy for this instead with a `policy(Budget).create?`?

Reply:
Again I don't know how to check the level of the activity at this point, because the budget does not yet exist. If I try to check the parent activity's level on the budget, none exists because the budget has not been created. If you pry the `record` at this point, you get:

`Budget(id: uuid, activity_id: uuid, budget_type: string, status: string, period_start_date: date, period_end_date: date, value: decimal)
`

Reply:
Ok. Could we make the assumption that if a user can create an activity for a given level (using `policy(Activity).create?`) then that user can also create budgets and transactions?

I'm not aware that we'll have user groups that can create particular parts of activities and not others.

Reply:
OK I will try that